app/jobs/cancel_reservations_job.rb

```diff
@@ -0,0 +1,23 @@
+class CancelReservationsJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    pending_reservations.each do |reservation|
+      cancel_reservation(reservation) if not_confirmed_on_time?(reservation)
+    end
+  end
+
+  private
+
+  def not_confirmed_on_time?(reservation)
+    reservation.seance.start_time <= 30.minutes.from_now
+  end
+
+  def pending_reservations
+    Reservation.where(status: :reserved)
+  end
```

Comment:
Why not query for reservations that have `status` reserved and screenings that start in 30 minutes or less?
```ruby
Reservation.joins(:seance).where(status: :reserved).where(seance: {start_time: ..30.minutes.from_now})
```

Something like that would probably work, it joins with reservation seances and then takes only those reservations for which seances starts in 30 minutes or less. That way you will take only those reservations that you should be cancelling, no all reservation in reserved states - after all, there will be much more of those.

Reply:
I made the changes and I'm using this query now. True, it was unnecessary taking all thous reservations.